From 9a87f6414274fd31c5e932bd0d5c6b7c7b455dfd Mon Sep 17 00:00:00 2001
From: user <Your Email>
Date: Mon, 16 Jun 2014 16:39:46 +0200
Subject: [PATCH 4/4] add sdcard config boards

---
 boards.cfg |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/boards.cfg b/boards.cfg
index 1cfeba6..ef0ad6d 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -293,6 +293,11 @@ mx6s_icore_512MB              arm         armv7       mx6_icore       engicam
 mx6dl_icore                   arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=512,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
 mx6d_icore                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=512,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
 mx6q_icore                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=1024,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
+mx6s_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=256,SYS_NOSMP="nosmp"
+mx6s_icore_512MB_sd              arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
+mx6dl_icore_sd                   arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
+mx6d_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
+mx6q_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=1024,SYS_NOSMP="nosmp"
 eco5pk                       arm         armv7       eco5pk              8dtech         omap3
 nitrogen6dl                  arm         armv7       nitrogen6x          boundary       mx6		nitrogen6x:IMX_CONFIG=board/boundary/nitrogen6x/nitrogen6dl.cfg,MX6DL,DDR_MB=1024
 nitrogen6dl2g                arm         armv7       nitrogen6x          boundary       mx6		nitrogen6x:IMX_CONFIG=board/boundary/nitrogen6x/nitrogen6dl2g.cfg,MX6DL,DDR_MB=2048
-- 
1.7.9.5

