diff --git a/include/configs/mx6_icore_common.h b/include/configs/mx6_icore_common.h
index 78496d0..1e219b9 100644
--- a/include/configs/mx6_icore_common.h
+++ b/include/configs/mx6_icore_common.h
@@ -150,6 +150,18 @@
 	 *     mtd3: 1M      (dtb)
 	 *     mtd4: left    (rootfs)
 	 */
+#ifdef CONFIG_MX6SOLO
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	CONFIG_MFG_ENV_SETTINGS \
+	"fdt_addr=0x18000000\0" \
+	"fdt_high=0xffffffff\0"	  \
+	"bootargs=console=" CONFIG_CONSOLE_DEV ",115200 ubi.mtd=4 "  \
+		"root=ubi0:rootfs rootfstype=ubifs cma=96MB gpumem=16MB video=mxcfb0:dev=lcd,Amp-WD "		     \
+		"mtdparts=gpmi-nand:4m(boot),8m(kernel),1m(dtb),-(rootfs)\0"\
+	"bootcmd=nand read ${loadaddr} 0x400000 0x700000;"\
+		"nand read ${fdt_addr} 0xc00000 0x100000;"\
+		"bootm ${loadaddr} - ${fdt_addr}\0"
+#else
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	CONFIG_MFG_ENV_SETTINGS \
 	"fdt_addr=0x18000000\0" \
@@ -160,6 +172,7 @@
 	"bootcmd=nand read ${loadaddr} 0x400000 0x700000;"\
 		"nand read ${fdt_addr} 0xc00000 0x100000;"\
 		"bootm ${loadaddr} - ${fdt_addr}\0"
+#endif
 
 #elif defined(CONFIG_SYS_BOOT_SATA)
 
-- 
1.7.9.5

