diff --git a/include/configs/common_parameter.h b/include/configs/common_parameter.h
index ae2ea56..ab4d918 100644
--- a/include/configs/common_parameter.h
+++ b/include/configs/common_parameter.h
@@ -16,6 +16,14 @@
 	#define CONFIG_NETMASK		255.255.255.0
 	#define CONFIG_ETHADDR		60:aa:60:aa:60:aa
 
+	#if defined(CONFIG_SYS_BOOT_NAND)
+		#define CONFIG_BOOTCMD	"bootcmd=run bootcmd_ubi\0"
+	#elif defined(CONFIG_SYS_BOOT_SATA)
+		#define CONFIG_BOOTCMD	"bootcmd=run bootcmd_ubi\0"
+	#else
+		#define CONFIG_BOOTCMD	"bootcmd=run bootcmd_mmc\0"
+	#endif
+
 	/* Common parameter
 	 * For all modules SODIMM and QSEVEN
 	 */
@@ -39,7 +47,7 @@
 		"bootcmd_ubi=run bootargs; run bootargs_ubi; nand read ${loadaddr} 0x400000 0x700000;nand read ${fdt_addr} 0xc00000 0x100000;bootm ${loadaddr} - ${fdt_addr}\0" 		\
 		"bootargs_ubi=setenv bootargs ${bootargs} ubi.mtd=3 root=ubi0:rootfs rootfstype=ubifs\0"				\
 		"video_type=mxcfb0:dev=lcd\0"		\
-		"bootcmd=run bootcmd_ubi\0" 		\
+		CONFIG_BOOTCMD 				\
 		"erase=nand erase 1c0000 4000\0"	\
 		"board=SK.RES\0"			\
 
@@ -47,7 +55,7 @@
 	 * Customized parameter for QSEVEN RQS modules
 	 */
 	#define	EXTRA_ENV_SETTINGS_RQS 			\
-		COMMON_PARAMETER 			\	
+		COMMON_PARAMETER 			\
 		"bootcmd_emmc=run bootargs; run bootargs_emmc; mmc dev 1; mmc read ${loadaddr} 0x800 0x2000;bootm\0"					\
 		"bootargs_emmc=setenv bootargs ${bootargs} root=/dev/mmcblk0p1 rootwait rw\0" 								\
 		"video_type=mxcfb0:dev=ldb,LDB-XGA,if=RGB666 video=mxcfb0:dev=hdmi,1920x1080M@60,if=RGB24\0"						\
-- 
1.9.1

