diff --git a/arch/arm/include/asm/arch-mx6/mx6dl_pins.h b/arch/arm/include/asm/arch-mx6/mx6dl_pins.h
index 3dd3bfe..3ae8b0b 100644
--- a/arch/arm/include/asm/arch-mx6/mx6dl_pins.h
+++ b/arch/arm/include/asm/arch-mx6/mx6dl_pins.h
@@ -190,7 +190,7 @@ enum {
 	MX6_PAD_CSI0_VSYNC__GPIO_5_21                          = IOMUX_PAD(0x03AC, 0x0098, 5, 0x0000, 0, 0),
 	MX6_PAD_CSI0_VSYNC__MMDC_MMDC_DEBUG_32                 = IOMUX_PAD(0x03AC, 0x0098, 6, 0x0000, 0, 0),
 	MX6_PAD_CSI0_VSYNC__SIMBA_TRACE_0                      = IOMUX_PAD(0x03AC, 0x0098, 7, 0x0000, 0, 0),
-	MX6_PAD_DI0_DISP_CLK__IPU1_DI0_DISP_CLK                = IOMUX_PAD(0x03B0, 0x009C, 0, 0x0000, 0, 0),
+	MX6_PAD_DI0_DISP_CLK__IPU1_DI0_DISP_CLK                = IOMUX_PAD(0x03B0, 0x009C, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DI0_DISP_CLK__LCDIF_CLK                        = IOMUX_PAD(0x03B0, 0x009C, 1, 0x0000, 0, 0),
 	MX6_PAD_DI0_DISP_CLK__MIPI_CORE_DPHY_TEST_OUT_28       = IOMUX_PAD(0x03B0, 0x009C, 3, 0x0000, 0, 0),
 	MX6_PAD_DI0_DISP_CLK__SDMA_DEBUG_CORE_STATE_0          = IOMUX_PAD(0x03B0, 0x009C, 4, 0x0000, 0, 0),
@@ -198,7 +198,7 @@ enum {
 	MX6_PAD_DI0_DISP_CLK__MMDC_MMDC_DEBUG_0                = IOMUX_PAD(0x03B0, 0x009C, 6, 0x0000, 0, 0),
 	MX6_PAD_DI0_DISP_CLK__TPSMP_HDATA_DIR                  = IOMUX_PAD(0x03B0, 0x009C, 7, 0x0000, 0, 0),
 	MX6_PAD_DI0_DISP_CLK__LCDIF_WR_RWN                     = IOMUX_PAD(0x03B0, 0x009C, 8, 0x0000, 0, 0),
-	MX6_PAD_DI0_PIN15__IPU1_DI0_PIN15                      = IOMUX_PAD(0x03B4, 0x00A0, 0, 0x0000, 0, 0),
+	MX6_PAD_DI0_PIN15__IPU1_DI0_PIN15                      = IOMUX_PAD(0x03B4, 0x00A0, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DI0_PIN15__LCDIF_ENABLE                        = IOMUX_PAD(0x03B4, 0x00A0, 1, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN15__AUDMUX_AUD6_TXC                     = IOMUX_PAD(0x03B4, 0x00A0, 2, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN15__MIPI_CORE_DPHY_TEST_OUT_29          = IOMUX_PAD(0x03B4, 0x00A0, 3, 0x0000, 0, 0),
@@ -207,7 +207,7 @@ enum {
 	MX6_PAD_DI0_PIN15__MMDC_MMDC_DEBUG_1                   = IOMUX_PAD(0x03B4, 0x00A0, 6, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN15__PL301_SIM_MX6_PER1_HSIZE_0        = IOMUX_PAD(0x03B4, 0x00A0, 7, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN15__LCDIF_RD_E                          = IOMUX_PAD(0x03B4, 0x00A0, 8, 0x0000, 0, 0),
-	MX6_PAD_DI0_PIN2__IPU1_DI0_PIN2                        = IOMUX_PAD(0x03B8, 0x00A4, 0, 0x0000, 0, 0),
+	MX6_PAD_DI0_PIN2__IPU1_DI0_PIN2                        = IOMUX_PAD(0x03B8, 0x00A4, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DI0_PIN2__LCDIF_HSYNC                          = IOMUX_PAD(0x03B8, 0x00A4, 1, 0x08D8, 0, 0),
 	MX6_PAD_DI0_PIN2__AUDMUX_AUD6_TXD                      = IOMUX_PAD(0x03B8, 0x00A4, 2, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN2__MIPI_CORE_DPHY_TEST_OUT_30           = IOMUX_PAD(0x03B8, 0x00A4, 3, 0x0000, 0, 0),
@@ -216,7 +216,7 @@ enum {
 	MX6_PAD_DI0_PIN2__MMDC_MMDC_DEBUG_2                    = IOMUX_PAD(0x03B8, 0x00A4, 6, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN2__PL301_SIM_MX6_PER1_HADDR_9         = IOMUX_PAD(0x03B8, 0x00A4, 7, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN2__LCDIF_RS                             = IOMUX_PAD(0x03B8, 0x00A4, 8, 0x0000, 0, 0),
-	MX6_PAD_DI0_PIN3__IPU1_DI0_PIN3                        = IOMUX_PAD(0x03BC, 0x00A8, 0, 0x0000, 0, 0),
+	MX6_PAD_DI0_PIN3__IPU1_DI0_PIN3                        = IOMUX_PAD(0x03BC, 0x00A8, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DI0_PIN3__LCDIF_VSYNC                          = IOMUX_PAD(0x03BC, 0x00A8, 1, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN3__AUDMUX_AUD6_TXFS                     = IOMUX_PAD(0x03BC, 0x00A8, 2, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN3__MIPI_CORE_DPHY_TEST_OUT_31           = IOMUX_PAD(0x03BC, 0x00A8, 3, 0x0000, 0, 0),
@@ -234,7 +234,7 @@ enum {
 	MX6_PAD_DI0_PIN4__MMDC_MMDC_DEBUG_4                    = IOMUX_PAD(0x03C0, 0x00AC, 6, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN4__PL301_SIM_MX6_PER1_HADDR_11        = IOMUX_PAD(0x03C0, 0x00AC, 7, 0x0000, 0, 0),
 	MX6_PAD_DI0_PIN4__LCDIF_RESET                          = IOMUX_PAD(0x03C0, 0x00AC, 8, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT0__IPU1_DISP0_DAT_0                   = IOMUX_PAD(0x03C4, 0x00B0, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT0__IPU1_DISP0_DAT_0                   = IOMUX_PAD(0x03C4, 0x00B0, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT0__LCDIF_DAT_0                        = IOMUX_PAD(0x03C4, 0x00B0, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT0__ECSPI3_SCLK                        = IOMUX_PAD(0x03C4, 0x00B0, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT0__USDHC1_USDHC_DEBUG_0               = IOMUX_PAD(0x03C4, 0x00B0, 3, 0x0000, 0, 0),
@@ -242,7 +242,7 @@ enum {
 	MX6_PAD_DISP0_DAT0__GPIO_4_21                          = IOMUX_PAD(0x03C4, 0x00B0, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT0__MMDC_MMDC_DEBUG_5                  = IOMUX_PAD(0x03C4, 0x00B0, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT0__PL301_SIM_MX6_PER1_HSIZE_1       = IOMUX_PAD(0x03C4, 0x00B0, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT1__IPU1_DISP0_DAT_1                   = IOMUX_PAD(0x03C8, 0x00B4, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT1__IPU1_DISP0_DAT_1                   = IOMUX_PAD(0x03C8, 0x00B4, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT1__LCDIF_DAT_1                        = IOMUX_PAD(0x03C8, 0x00B4, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT1__ECSPI3_MOSI                        = IOMUX_PAD(0x03C8, 0x00B4, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT1__USDHC1_USDHC_DEBUG_1               = IOMUX_PAD(0x03C8, 0x00B4, 3, 0x0000, 0, 0),
@@ -250,41 +250,41 @@ enum {
 	MX6_PAD_DISP0_DAT1__GPIO_4_22                          = IOMUX_PAD(0x03C8, 0x00B4, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT1__MMDC_MMDC_DEBUG_6                  = IOMUX_PAD(0x03C8, 0x00B4, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT1__PL301_SIM_MX6_PER1_HADDR_12      = IOMUX_PAD(0x03C8, 0x00B4, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT10__IPU1_DISP0_DAT_10                 = IOMUX_PAD(0x03CC, 0x00B8, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT10__IPU1_DISP0_DAT_10                 = IOMUX_PAD(0x03CC, 0x00B8, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT10__LCDIF_DAT_10                      = IOMUX_PAD(0x03CC, 0x00B8, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT10__USDHC1_USDHC_DEBUG_6              = IOMUX_PAD(0x03CC, 0x00B8, 3, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT10__SDMA_DEBUG_EVENT_CHANNEL_3        = IOMUX_PAD(0x03CC, 0x00B8, 4, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT10__GPIO_4_31                         = IOMUX_PAD(0x03CC, 0x00B8, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT10__MMDC_MMDC_DEBUG_15                = IOMUX_PAD(0x03CC, 0x00B8, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT10__PL301_SIM_MX6_PER1_HADDR_21     = IOMUX_PAD(0x03CC, 0x00B8, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT11__IPU1_DISP0_DAT_11                 = IOMUX_PAD(0x03D0, 0x00BC, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT11__IPU1_DISP0_DAT_11                 = IOMUX_PAD(0x03D0, 0x00BC, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT11__LCDIF_DAT_11                      = IOMUX_PAD(0x03D0, 0x00BC, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT11__USDHC1_USDHC_DEBUG_7              = IOMUX_PAD(0x03D0, 0x00BC, 3, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT11__SDMA_DEBUG_EVENT_CHANNEL_4        = IOMUX_PAD(0x03D0, 0x00BC, 4, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT11__GPIO_5_5                          = IOMUX_PAD(0x03D0, 0x00BC, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT11__MMDC_MMDC_DEBUG_16                = IOMUX_PAD(0x03D0, 0x00BC, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT11__PL301_SIM_MX6_PER1_HADDR_22     = IOMUX_PAD(0x03D0, 0x00BC, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT12__IPU1_DISP0_DAT_12                 = IOMUX_PAD(0x03D4, 0x00C0, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT12__IPU1_DISP0_DAT_12                 = IOMUX_PAD(0x03D4, 0x00C0, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT12__LCDIF_DAT_12                      = IOMUX_PAD(0x03D4, 0x00C0, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT12__SDMA_DEBUG_EVENT_CHANNEL_5        = IOMUX_PAD(0x03D4, 0x00C0, 4, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT12__GPIO_5_6                          = IOMUX_PAD(0x03D4, 0x00C0, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT12__MMDC_MMDC_DEBUG_17                = IOMUX_PAD(0x03D4, 0x00C0, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT12__PL301_SIM_MX6_PER1_HADDR_23     = IOMUX_PAD(0x03D4, 0x00C0, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT13__IPU1_DISP0_DAT_13                 = IOMUX_PAD(0x03D8, 0x00C4, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT13__IPU1_DISP0_DAT_13                 = IOMUX_PAD(0x03D8, 0x00C4, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT13__LCDIF_DAT_13                      = IOMUX_PAD(0x03D8, 0x00C4, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT13__AUDMUX_AUD5_RXFS                  = IOMUX_PAD(0x03D8, 0x00C4, 3, 0x07BC, 0, 0),
 	MX6_PAD_DISP0_DAT13__SDMA_DEBUG_EVT_CHN_LINES_0        = IOMUX_PAD(0x03D8, 0x00C4, 4, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT13__GPIO_5_7                          = IOMUX_PAD(0x03D8, 0x00C4, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT13__MMDC_MMDC_DEBUG_18                = IOMUX_PAD(0x03D8, 0x00C4, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT13__PL301_SIM_MX6_PER1_HADDR_24     = IOMUX_PAD(0x03D8, 0x00C4, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT14__IPU1_DISP0_DAT_14                 = IOMUX_PAD(0x03DC, 0x00C8, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT14__IPU1_DISP0_DAT_14                 = IOMUX_PAD(0x03DC, 0x00C8, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT14__LCDIF_DAT_14                      = IOMUX_PAD(0x03DC, 0x00C8, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT14__AUDMUX_AUD5_RXC                   = IOMUX_PAD(0x03DC, 0x00C8, 3, 0x07B8, 0, 0),
 	MX6_PAD_DISP0_DAT14__SDMA_DEBUG_EVT_CHN_LINES_1        = IOMUX_PAD(0x03DC, 0x00C8, 4, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT14__GPIO_5_8                          = IOMUX_PAD(0x03DC, 0x00C8, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT14__MMDC_MMDC_DEBUG_19                = IOMUX_PAD(0x03DC, 0x00C8, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT14__PL301_SIM_MX6_PER1_HSIZE_2      = IOMUX_PAD(0x03DC, 0x00C8, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT15__IPU1_DISP0_DAT_15                 = IOMUX_PAD(0x03E0, 0x00CC, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT15__IPU1_DISP0_DAT_15                 = IOMUX_PAD(0x03E0, 0x00CC, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT15__LCDIF_DAT_15                      = IOMUX_PAD(0x03E0, 0x00CC, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT15__ECSPI1_SS1                        = IOMUX_PAD(0x03E0, 0x00CC, 2, 0x07E8, 0, 0),
 	MX6_PAD_DISP0_DAT15__ECSPI2_SS1                        = IOMUX_PAD(0x03E0, 0x00CC, 3, 0x0804, 0, 0),
@@ -292,7 +292,7 @@ enum {
 	MX6_PAD_DISP0_DAT15__GPIO_5_9                          = IOMUX_PAD(0x03E0, 0x00CC, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT15__MMDC_MMDC_DEBUG_20                = IOMUX_PAD(0x03E0, 0x00CC, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT15__PL301_SIM_MX6_PER1_HADDR_25     = IOMUX_PAD(0x03E0, 0x00CC, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT16__IPU1_DISP0_DAT_16                 = IOMUX_PAD(0x03E4, 0x00D0, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT16__IPU1_DISP0_DAT_16                 = IOMUX_PAD(0x03E4, 0x00D0, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT16__LCDIF_DAT_16                      = IOMUX_PAD(0x03E4, 0x00D0, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT16__ECSPI2_MOSI                       = IOMUX_PAD(0x03E4, 0x00D0, 2, 0x07FC, 1, 0),
 	MX6_PAD_DISP0_DAT16__AUDMUX_AUD5_TXC                   = IOMUX_PAD(0x03E4, 0x00D0, 3, 0x07C0, 0, 0),
@@ -300,7 +300,7 @@ enum {
 	MX6_PAD_DISP0_DAT16__GPIO_5_10                         = IOMUX_PAD(0x03E4, 0x00D0, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT16__MMDC_MMDC_DEBUG_21                = IOMUX_PAD(0x03E4, 0x00D0, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT16__PL301_SIM_MX6_PER1_HADDR_26     = IOMUX_PAD(0x03E4, 0x00D0, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT17__IPU1_DISP0_DAT_17                 = IOMUX_PAD(0x03E8, 0x00D4, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT17__IPU1_DISP0_DAT_17                 = IOMUX_PAD(0x03E8, 0x00D4, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT17__LCDIF_DAT_17                      = IOMUX_PAD(0x03E8, 0x00D4, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT17__ECSPI2_MISO                       = IOMUX_PAD(0x03E8, 0x00D4, 2, 0x07F8, 1, 0),
 	MX6_PAD_DISP0_DAT17__AUDMUX_AUD5_TXD                   = IOMUX_PAD(0x03E8, 0x00D4, 3, 0x07B4, 0, 0),
@@ -308,7 +308,7 @@ enum {
 	MX6_PAD_DISP0_DAT17__GPIO_5_11                         = IOMUX_PAD(0x03E8, 0x00D4, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT17__MMDC_MMDC_DEBUG_22                = IOMUX_PAD(0x03E8, 0x00D4, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT17__PL301_SIM_MX6_PER1_HADDR_27     = IOMUX_PAD(0x03E8, 0x00D4, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT18__IPU1_DISP0_DAT_18                 = IOMUX_PAD(0x03EC, 0x00D8, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT18__IPU1_DISP0_DAT_18                 = IOMUX_PAD(0x03EC, 0x00D8, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT18__LCDIF_DAT_18                      = IOMUX_PAD(0x03EC, 0x00D8, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT18__ECSPI2_SS0                        = IOMUX_PAD(0x03EC, 0x00D8, 2, 0x0800, 1, 0),
 	MX6_PAD_DISP0_DAT18__AUDMUX_AUD5_TXFS                  = IOMUX_PAD(0x03EC, 0x00D8, 3, 0x07C4, 0, 0),
@@ -316,7 +316,7 @@ enum {
 	MX6_PAD_DISP0_DAT18__GPIO_5_12                         = IOMUX_PAD(0x03EC, 0x00D8, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT18__MMDC_MMDC_DEBUG_23                = IOMUX_PAD(0x03EC, 0x00D8, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT18__WEIM_WEIM_CS_2                    = IOMUX_PAD(0x03EC, 0x00D8, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT19__IPU1_DISP0_DAT_19                 = IOMUX_PAD(0x03F0, 0x00DC, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT19__IPU1_DISP0_DAT_19                 = IOMUX_PAD(0x03F0, 0x00DC, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT19__LCDIF_DAT_19                      = IOMUX_PAD(0x03F0, 0x00DC, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT19__ECSPI2_SCLK                       = IOMUX_PAD(0x03F0, 0x00DC, 2, 0x07F4, 1, 0),
 	MX6_PAD_DISP0_DAT19__AUDMUX_AUD5_RXD                   = IOMUX_PAD(0x03F0, 0x00DC, 3, 0x07B0, 0, 0),
@@ -324,7 +324,7 @@ enum {
 	MX6_PAD_DISP0_DAT19__GPIO_5_13                         = IOMUX_PAD(0x03F0, 0x00DC, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT19__MMDC_MMDC_DEBUG_24                = IOMUX_PAD(0x03F0, 0x00DC, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT19__WEIM_WEIM_CS_3                    = IOMUX_PAD(0x03F0, 0x00DC, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT2__IPU1_DISP0_DAT_2                   = IOMUX_PAD(0x03F4, 0x00E0, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT2__IPU1_DISP0_DAT_2                   = IOMUX_PAD(0x03F4, 0x00E0, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT2__LCDIF_DAT_2                        = IOMUX_PAD(0x03F4, 0x00E0, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT2__ECSPI3_MISO                        = IOMUX_PAD(0x03F4, 0x00E0, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT2__USDHC1_USDHC_DEBUG_2               = IOMUX_PAD(0x03F4, 0x00E0, 3, 0x0000, 0, 0),
@@ -332,7 +332,7 @@ enum {
 	MX6_PAD_DISP0_DAT2__GPIO_4_23                          = IOMUX_PAD(0x03F4, 0x00E0, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT2__MMDC_MMDC_DEBUG_7                  = IOMUX_PAD(0x03F4, 0x00E0, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT2__PL301_SIM_MX6_PER1_HADDR_13      = IOMUX_PAD(0x03F4, 0x00E0, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT20__IPU1_DISP0_DAT_20                 = IOMUX_PAD(0x03F8, 0x00E4, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT20__IPU1_DISP0_DAT_20                 = IOMUX_PAD(0x03F8, 0x00E4, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT20__LCDIF_DAT_20                      = IOMUX_PAD(0x03F8, 0x00E4, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT20__ECSPI1_SCLK                       = IOMUX_PAD(0x03F8, 0x00E4, 2, 0x07D8, 1, 0),
 	MX6_PAD_DISP0_DAT20__AUDMUX_AUD4_TXC                   = IOMUX_PAD(0x03F8, 0x00E4, 3, 0x07A8, 0, 0),
@@ -340,7 +340,7 @@ enum {
 	MX6_PAD_DISP0_DAT20__GPIO_5_14                         = IOMUX_PAD(0x03F8, 0x00E4, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT20__MMDC_MMDC_DEBUG_25                = IOMUX_PAD(0x03F8, 0x00E4, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT20__PL301_SIM_MX6_PER1_HADDR_28     = IOMUX_PAD(0x03F8, 0x00E4, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT21__IPU1_DISP0_DAT_21                 = IOMUX_PAD(0x03FC, 0x00E8, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT21__IPU1_DISP0_DAT_21                 = IOMUX_PAD(0x03FC, 0x00E8, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT21__LCDIF_DAT_21                      = IOMUX_PAD(0x03FC, 0x00E8, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT21__ECSPI1_MOSI                       = IOMUX_PAD(0x03FC, 0x00E8, 2, 0x07E0, 1, 0),
 	MX6_PAD_DISP0_DAT21__AUDMUX_AUD4_TXD                   = IOMUX_PAD(0x03FC, 0x00E8, 3, 0x079C, 0, 0),
@@ -348,7 +348,7 @@ enum {
 	MX6_PAD_DISP0_DAT21__GPIO_5_15                         = IOMUX_PAD(0x03FC, 0x00E8, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT21__MMDC_MMDC_DEBUG_26                = IOMUX_PAD(0x03FC, 0x00E8, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT21__PL301_SIM_MX6_PER1_HADDR_29     = IOMUX_PAD(0x03FC, 0x00E8, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT22__IPU1_DISP0_DAT_22                 = IOMUX_PAD(0x0400, 0x00EC, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT22__IPU1_DISP0_DAT_22                 = IOMUX_PAD(0x0400, 0x00EC, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT22__LCDIF_DAT_22                      = IOMUX_PAD(0x0400, 0x00EC, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT22__ECSPI1_MISO                       = IOMUX_PAD(0x0400, 0x00EC, 2, 0x07DC, 1, 0),
 	MX6_PAD_DISP0_DAT22__AUDMUX_AUD4_TXFS                  = IOMUX_PAD(0x0400, 0x00EC, 3, 0x07AC, 0, 0),
@@ -356,7 +356,7 @@ enum {
 	MX6_PAD_DISP0_DAT22__GPIO_5_16                         = IOMUX_PAD(0x0400, 0x00EC, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT22__MMDC_MMDC_DEBUG_27                = IOMUX_PAD(0x0400, 0x00EC, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT22__PL301_SIM_MX6_PER1_HADDR_30     = IOMUX_PAD(0x0400, 0x00EC, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT23__IPU1_DISP0_DAT_23                 = IOMUX_PAD(0x0404, 0x00F0, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT23__IPU1_DISP0_DAT_23                 = IOMUX_PAD(0x0404, 0x00F0, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT23__LCDIF_DAT_23                      = IOMUX_PAD(0x0404, 0x00F0, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT23__ECSPI1_SS0                        = IOMUX_PAD(0x0404, 0x00F0, 2, 0x07E4, 1, 0),
 	MX6_PAD_DISP0_DAT23__AUDMUX_AUD4_RXD                   = IOMUX_PAD(0x0404, 0x00F0, 3, 0x0798, 0, 0),
@@ -364,7 +364,7 @@ enum {
 	MX6_PAD_DISP0_DAT23__GPIO_5_17                         = IOMUX_PAD(0x0404, 0x00F0, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT23__MMDC_MMDC_DEBUG_28                = IOMUX_PAD(0x0404, 0x00F0, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT23__PL301_SIM_MX6_PER1_HADDR_31     = IOMUX_PAD(0x0404, 0x00F0, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT3__IPU1_DISP0_DAT_3                   = IOMUX_PAD(0x0408, 0x00F4, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT3__IPU1_DISP0_DAT_3                   = IOMUX_PAD(0x0408, 0x00F4, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT3__LCDIF_DAT_3                        = IOMUX_PAD(0x0408, 0x00F4, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT3__ECSPI3_SS0                         = IOMUX_PAD(0x0408, 0x00F4, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT3__USDHC1_USDHC_DEBUG_3               = IOMUX_PAD(0x0408, 0x00F4, 3, 0x0000, 0, 0),
@@ -372,7 +372,7 @@ enum {
 	MX6_PAD_DISP0_DAT3__GPIO_4_24                          = IOMUX_PAD(0x0408, 0x00F4, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT3__MMDC_MMDC_DEBUG_8                  = IOMUX_PAD(0x0408, 0x00F4, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT3__PL301_SIM_MX6_PER1_HADDR_14      = IOMUX_PAD(0x0408, 0x00F4, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT4__IPU1_DISP0_DAT_4                   = IOMUX_PAD(0x040C, 0x00F8, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT4__IPU1_DISP0_DAT_4                   = IOMUX_PAD(0x040C, 0x00F8, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT4__LCDIF_DAT_4                        = IOMUX_PAD(0x040C, 0x00F8, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT4__ECSPI3_SS1                         = IOMUX_PAD(0x040C, 0x00F8, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT4__USDHC1_USDHC_DEBUG_4               = IOMUX_PAD(0x040C, 0x00F8, 3, 0x0000, 0, 0),
@@ -380,7 +380,7 @@ enum {
 	MX6_PAD_DISP0_DAT4__GPIO_4_25                          = IOMUX_PAD(0x040C, 0x00F8, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT4__MMDC_MMDC_DEBUG_9                  = IOMUX_PAD(0x040C, 0x00F8, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT4__PL301_SIM_MX6_PER1_HADDR_15      = IOMUX_PAD(0x040C, 0x00F8, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT5__IPU1_DISP0_DAT_5                   = IOMUX_PAD(0x0410, 0x00FC, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT5__IPU1_DISP0_DAT_5                   = IOMUX_PAD(0x0410, 0x00FC, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT5__LCDIF_DAT_5                        = IOMUX_PAD(0x0410, 0x00FC, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT5__ECSPI3_SS2                         = IOMUX_PAD(0x0410, 0x00FC, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT5__AUDMUX_AUD6_RXFS                   = IOMUX_PAD(0x0410, 0x00FC, 3, 0x0000, 0, 0),
@@ -388,7 +388,7 @@ enum {
 	MX6_PAD_DISP0_DAT5__GPIO_4_26                          = IOMUX_PAD(0x0410, 0x00FC, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT5__MMDC_MMDC_DEBUG_10                 = IOMUX_PAD(0x0410, 0x00FC, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT5__PL301_SIM_MX6_PER1_HADDR_16      = IOMUX_PAD(0x0410, 0x00FC, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT6__IPU1_DISP0_DAT_6                   = IOMUX_PAD(0x0414, 0x0100, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT6__IPU1_DISP0_DAT_6                   = IOMUX_PAD(0x0414, 0x0100, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT6__LCDIF_DAT_6                        = IOMUX_PAD(0x0414, 0x0100, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT6__ECSPI3_SS3                         = IOMUX_PAD(0x0414, 0x0100, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT6__AUDMUX_AUD6_RXC                    = IOMUX_PAD(0x0414, 0x0100, 3, 0x0000, 0, 0),
@@ -396,7 +396,7 @@ enum {
 	MX6_PAD_DISP0_DAT6__GPIO_4_27                          = IOMUX_PAD(0x0414, 0x0100, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT6__MMDC_MMDC_DEBUG_11                 = IOMUX_PAD(0x0414, 0x0100, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT6__PL301_SIM_MX6_PER1_HADDR_17      = IOMUX_PAD(0x0414, 0x0100, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT7__IPU1_DISP0_DAT_7                   = IOMUX_PAD(0x0418, 0x0104, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT7__IPU1_DISP0_DAT_7                   = IOMUX_PAD(0x0418, 0x0104, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT7__LCDIF_DAT_7                        = IOMUX_PAD(0x0418, 0x0104, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT7__ECSPI3_RDY                         = IOMUX_PAD(0x0418, 0x0104, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT7__USDHC1_USDHC_DEBUG_5               = IOMUX_PAD(0x0418, 0x0104, 3, 0x0000, 0, 0),
@@ -404,7 +404,7 @@ enum {
 	MX6_PAD_DISP0_DAT7__GPIO_4_28                          = IOMUX_PAD(0x0418, 0x0104, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT7__MMDC_MMDC_DEBUG_12                 = IOMUX_PAD(0x0418, 0x0104, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT7__PL301_SIM_MX6_PER1_HADDR_18      = IOMUX_PAD(0x0418, 0x0104, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT8__IPU1_DISP0_DAT_8                   = IOMUX_PAD(0x041C, 0x0108, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT8__IPU1_DISP0_DAT_8                   = IOMUX_PAD(0x041C, 0x0108, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT8__LCDIF_DAT_8                        = IOMUX_PAD(0x041C, 0x0108, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT8__PWM1_PWMO                          = IOMUX_PAD(0x041C, 0x0108, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT8__WDOG1_WDOG_B                       = IOMUX_PAD(0x041C, 0x0108, 3, 0x0000, 0, 0),
@@ -412,7 +412,7 @@ enum {
 	MX6_PAD_DISP0_DAT8__GPIO_4_29                          = IOMUX_PAD(0x041C, 0x0108, 5, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT8__MMDC_MMDC_DEBUG_13                 = IOMUX_PAD(0x041C, 0x0108, 6, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT8__PL301_SIM_MX6_PER1_HADDR_19      = IOMUX_PAD(0x041C, 0x0108, 7, 0x0000, 0, 0),
-	MX6_PAD_DISP0_DAT9__IPU1_DISP0_DAT_9                   = IOMUX_PAD(0x0420, 0x010C, 0, 0x0000, 0, 0),
+	MX6_PAD_DISP0_DAT9__IPU1_DISP0_DAT_9                   = IOMUX_PAD(0x0420, 0x010C, 0, 0x0000, 0, PAD_CTL_DSE_120ohm),
 	MX6_PAD_DISP0_DAT9__LCDIF_DAT_9                        = IOMUX_PAD(0x0420, 0x010C, 1, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT9__PWM2_PWMO                          = IOMUX_PAD(0x0420, 0x010C, 2, 0x0000, 0, 0),
 	MX6_PAD_DISP0_DAT9__WDOG2_WDOG_B                       = IOMUX_PAD(0x0420, 0x010C, 3, 0x0000, 0, 0),
diff --git a/arch/powerpc/config.mk b/arch/powerpc/config.mk
old mode 100755
new mode 100644
diff --git a/board/engicam/mx6_icore/mx6_icore.c b/board/engicam/mx6_icore/mx6_icore.c
index dc35181..eb6664f 100644
--- a/board/engicam/mx6_icore/mx6_icore.c
+++ b/board/engicam/mx6_icore/mx6_icore.c
@@ -39,6 +39,7 @@
 #include <netdev.h>
 #include <asm/arch/sys_proto.h>
 #include <asm/arch/mxc_hdmi.h>
+#include "../../../drivers/video/mxcfb.h"
 #include <linux/fb.h>
 #include <ipu_pixfmt.h>
 
@@ -238,6 +239,7 @@ static int setup_pmic_voltages(void)
 #ifdef CONFIG_LDO_BYPASS_CHECK
 void ldo_mode_set(int ldo_bypass)
 {
+#ifdef TOLTO_ICORE
 	unsigned char value;
 	/* increase VDDARM/VDDSOC to support 1.2G chip */
 	if (check_1_2G()) {
@@ -266,6 +268,9 @@ void ldo_mode_set(int ldo_bypass)
 			return;
 		}
 	}
+#else
+	printf("ldo_mode_set removed \n");
+#endif
 }
 #endif
 #endif
@@ -472,6 +477,42 @@ int board_phy_config(struct phy_device *phydev)
 }
 
 #if defined(CONFIG_VIDEO_IPUV3)
+static iomux_v3_cfg_t const rgb_pads[] = {
+	MX6_PAD_DI0_DISP_CLK__IPU1_DI0_DISP_CLK,
+	MX6_PAD_DI0_PIN15__IPU1_DI0_PIN15,
+	MX6_PAD_DI0_PIN2__IPU1_DI0_PIN2,
+	MX6_PAD_DI0_PIN3__IPU1_DI0_PIN3,
+	MX6_PAD_DI0_PIN4__GPIO_4_20 | MUX_PAD_CTRL(ENET_PAD_CTRL),
+	MX6_PAD_DISP0_DAT0__IPU1_DISP0_DAT_0,
+	MX6_PAD_DISP0_DAT1__IPU1_DISP0_DAT_1,
+	MX6_PAD_DISP0_DAT2__IPU1_DISP0_DAT_2,
+	MX6_PAD_DISP0_DAT3__IPU1_DISP0_DAT_3,
+	MX6_PAD_DISP0_DAT4__IPU1_DISP0_DAT_4,
+	MX6_PAD_DISP0_DAT5__IPU1_DISP0_DAT_5,
+	MX6_PAD_DISP0_DAT6__IPU1_DISP0_DAT_6,
+	MX6_PAD_DISP0_DAT7__IPU1_DISP0_DAT_7,
+	MX6_PAD_DISP0_DAT8__IPU1_DISP0_DAT_8,
+	MX6_PAD_DISP0_DAT9__IPU1_DISP0_DAT_9,
+	MX6_PAD_DISP0_DAT10__IPU1_DISP0_DAT_10,
+	MX6_PAD_DISP0_DAT11__IPU1_DISP0_DAT_11,
+	MX6_PAD_DISP0_DAT12__IPU1_DISP0_DAT_12,
+	MX6_PAD_DISP0_DAT13__IPU1_DISP0_DAT_13,
+	MX6_PAD_DISP0_DAT14__IPU1_DISP0_DAT_14,
+	MX6_PAD_DISP0_DAT15__IPU1_DISP0_DAT_15,
+	MX6_PAD_DISP0_DAT16__IPU1_DISP0_DAT_16,
+	MX6_PAD_DISP0_DAT17__IPU1_DISP0_DAT_17,
+};
+
+#define RGB_BACKLIGHT_GP IMX_GPIO_NR(4, 20)
+
+static void enable_rgb(struct display_info_t const *dev)
+{
+	imx_iomux_v3_setup_multiple_pads(
+		rgb_pads,
+		 ARRAY_SIZE(rgb_pads));
+	gpio_direction_output(RGB_BACKLIGHT_GP, 1);
+}
+
 struct display_info_t {
 	int	bus;
 	int	addr;
@@ -504,6 +545,26 @@ static struct display_info_t const displays[] = {{
 	.addr	= 0,
 	.pixfmt	= IPU_PIX_FMT_RGB666,
 	.detect	= NULL,
+	.enable	= enable_rgb,
+	.mode	= {
+		.name           = "Amp-WD",
+		.refresh        = 60,
+		.xres           = 800,
+		.yres           = 480,
+		.pixclock       = 30000,
+		.left_margin    = 30,
+		.right_margin   = 30,
+		.upper_margin   = 5,
+		.lower_margin   = 5,
+		.hsync_len      = 64,
+		.vsync_len      = 20,
+		.sync           = FB_SYNC_CLK_LAT_FALL,
+		.vmode          = FB_VMODE_NONINTERLACED
+} }, {
+	.bus	= -1,
+	.addr	= 0,
+	.pixfmt	= IPU_PIX_FMT_RGB666,
+	.detect	= NULL,
 	.enable	= NULL,
 	.mode	= {
 		.name           = "Hannstar-XGA",
@@ -603,54 +664,54 @@ static void setup_iomux_backlight(void)
 static void setup_display(void)
 {
 	struct mxc_ccm_reg *mxc_ccm = (struct mxc_ccm_reg *)CCM_BASE_ADDR;
+	struct anatop_regs *anatop = (struct anatop_regs *)ANATOP_BASE_ADDR;
 	struct iomuxc *iomux = (struct iomuxc *)IOMUXC_BASE_ADDR;
 	int reg;
 
-	setup_iomux_backlight();
 	enable_ipu_clock();
 	imx_setup_hdmi();
-
-	/* Turn on LDB_DI0 and LDB_DI1 clocks */
-	reg = readl(&mxc_ccm->CCGR3);
-	reg |= MXC_CCM_CCGR3_LDB_DI0_MASK | MXC_CCM_CCGR3_LDB_DI1_MASK;
+	/* Turn on LDB0,IPU,IPU DI0 clocks */
+	reg = __raw_readl(&mxc_ccm->CCGR3);
+	reg |=  (MXC_CCM_CCGR3_LDB_DI0_MASK | 0xffff); //TBD MM MP:
+	printf("CCGR3 = 0x%x\n", reg);
 	writel(reg, &mxc_ccm->CCGR3);
 
-	/* Set LDB_DI0 and LDB_DI1 clk select to 3b'011 */
+	/* set PFD1_FRAC to 0x13 == 455 MHz (480*18)/0x13 */
+	writel(ANATOP_PFD_480_PFD1_FRAC_MASK, &anatop->pfd_480_clr);
+	writel(0x13<<ANATOP_PFD_480_PFD1_FRAC_SHIFT, &anatop->pfd_480_set);
+
+	/* set LDB0, LDB1 clk select to 011/011 */
 	reg = readl(&mxc_ccm->cs2cdr);
-	reg &= ~(MXC_CCM_CS2CDR_LDB_DI0_CLK_SEL_MASK |
-		 MXC_CCM_CS2CDR_LDB_DI1_CLK_SEL_MASK);
-	reg |= (3 << MXC_CCM_CS2CDR_LDB_DI0_CLK_SEL_OFFSET) |
-	       (3 << MXC_CCM_CS2CDR_LDB_DI1_CLK_SEL_OFFSET);
+	reg &= ~(MXC_CCM_CS2CDR_LDB_DI0_CLK_SEL_MASK
+		 |MXC_CCM_CS2CDR_LDB_DI1_CLK_SEL_MASK);
+	reg |= (3<<MXC_CCM_CS2CDR_LDB_DI0_CLK_SEL_OFFSET)
+	      |(3<<MXC_CCM_CS2CDR_LDB_DI1_CLK_SEL_OFFSET);
 	writel(reg, &mxc_ccm->cs2cdr);
 
 	reg = readl(&mxc_ccm->cscmr2);
-	reg |= MXC_CCM_CSCMR2_LDB_DI0_IPU_DIV | MXC_CCM_CSCMR2_LDB_DI1_IPU_DIV;
+	reg |= MXC_CCM_CSCMR2_LDB_DI0_IPU_DIV;
 	writel(reg, &mxc_ccm->cscmr2);
 
 	reg = readl(&mxc_ccm->chsccdr);
-	reg |= (CHSCCDR_CLK_SEL_LDB_DI0 <<
-		MXC_CCM_CHSCCDR_IPU1_DI0_CLK_SEL_OFFSET);
-	reg |= (CHSCCDR_CLK_SEL_LDB_DI0 <<
-		MXC_CCM_CHSCCDR_IPU1_DI1_CLK_SEL_OFFSET);
+	reg |= (CHSCCDR_CLK_SEL_LDB_DI0
+		<<MXC_CCM_CHSCCDR_IPU1_DI0_CLK_SEL_OFFSET);
 	writel(reg, &mxc_ccm->chsccdr);
 
-	reg = IOMUXC_GPR2_DI1_VS_POLARITY_ACTIVE_LOW |
-	      IOMUXC_GPR2_DI0_VS_POLARITY_ACTIVE_LOW |
-	      IOMUXC_GPR2_BIT_MAPPING_CH1_SPWG |
-	      IOMUXC_GPR2_DATA_WIDTH_CH1_18BIT |
-	      IOMUXC_GPR2_BIT_MAPPING_CH0_SPWG |
-	      IOMUXC_GPR2_DATA_WIDTH_CH0_18BIT |
-	      IOMUXC_GPR2_LVDS_CH0_MODE_ENABLED_DI0 |
-	      IOMUXC_GPR2_LVDS_CH1_MODE_DISABLED;
+	reg = IOMUXC_GPR2_BGREF_RRMODE_EXTERNAL_RES
+	     |IOMUXC_GPR2_DI1_VS_POLARITY_ACTIVE_HIGH
+	     |IOMUXC_GPR2_DI0_VS_POLARITY_ACTIVE_LOW
+	     |IOMUXC_GPR2_BIT_MAPPING_CH1_SPWG
+	     |IOMUXC_GPR2_DATA_WIDTH_CH1_18BIT
+	     |IOMUXC_GPR2_BIT_MAPPING_CH0_SPWG
+	     |IOMUXC_GPR2_DATA_WIDTH_CH0_18BIT
+	     |IOMUXC_GPR2_LVDS_CH1_MODE_DISABLED
+	     |IOMUXC_GPR2_LVDS_CH0_MODE_ENABLED_DI0;
 	writel(reg, &iomux->gpr[2]);
 
 	reg = readl(&iomux->gpr[3]);
-	reg &= ~(IOMUXC_GPR3_LVDS0_MUX_CTL_MASK |
-		 IOMUXC_GPR3_HDMI_MUX_CTL_MASK);
-	reg |= (IOMUXC_GPR3_MUX_SRC_IPU1_DI0 <<
-		IOMUXC_GPR3_LVDS0_MUX_CTL_OFFSET) |
-	       (IOMUXC_GPR3_MUX_SRC_IPU1_DI0 <<
-		IOMUXC_GPR3_HDMI_MUX_CTL_OFFSET);
+	reg = (reg & ~IOMUXC_GPR3_LVDS0_MUX_CTL_MASK)
+	    | (IOMUXC_GPR3_MUX_SRC_IPU1_DI0
+	       <<IOMUXC_GPR3_LVDS0_MUX_CTL_OFFSET);
 	writel(reg, &iomux->gpr[3]);
 }
 #endif /* CONFIG_VIDEO_IPUV3 */
diff --git a/board/exmeritus/hww1u1a/tlb.c b/board/exmeritus/hww1u1a/tlb.c
old mode 100755
new mode 100644
diff --git a/board/fads/Makefile b/board/fads/Makefile
old mode 100755
new mode 100644
diff --git a/common/env_sata.c b/common/env_sata.c
old mode 100644
new mode 100755
diff --git a/tools/Makefile b/tools/Makefile
index 889c897..f0ad080 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -140,6 +140,9 @@ endif
 ifeq ($(VENDOR),freescale)
 LOGO_BMP= logos/freescale.bmp
 endif
+ifeq ($(VENDOR),engicam)
+LOGO_BMP= logos/Engicam.bmp
+endif
 ifeq ($(VENDOR),ronetix)
 LOGO_BMP= logos/ronetix.bmp
 endif

1.9.1

