diff --git a/board/engicam/mx6_icore/mx6_icore.c b/board/engicam/mx6_icore/mx6_icore.c
index e2a36aa..9ba6d04 100644
--- a/board/engicam/mx6_icore/mx6_icore.c
+++ b/board/engicam/mx6_icore/mx6_icore.c
@@ -488,7 +488,6 @@ static iomux_v3_cfg_t const rgb_pads[] = {
 	MX6_PAD_DI0_PIN15__IPU1_DI0_PIN15,
 	MX6_PAD_DI0_PIN2__IPU1_DI0_PIN2,
 	MX6_PAD_DI0_PIN3__IPU1_DI0_PIN3,
-	MX6_PAD_DI0_PIN4__GPIO_4_20 | MUX_PAD_CTRL(ENET_PAD_CTRL),
 	MX6_PAD_DISP0_DAT0__IPU1_DISP0_DAT_0,
 	MX6_PAD_DISP0_DAT1__IPU1_DISP0_DAT_1,
 	MX6_PAD_DISP0_DAT2__IPU1_DISP0_DAT_2,
@@ -509,14 +508,11 @@ static iomux_v3_cfg_t const rgb_pads[] = {
 	MX6_PAD_DISP0_DAT17__IPU1_DISP0_DAT_17,
 };
 
-#define RGB_BACKLIGHT_GP IMX_GPIO_NR(4, 20)
-
 static void enable_rgb(struct display_info_t const *dev)
 {
 	imx_iomux_v3_setup_multiple_pads(
 		rgb_pads,
 		 ARRAY_SIZE(rgb_pads));
-	gpio_direction_output(RGB_BACKLIGHT_GP, 1);
 }
 
 struct display_info_t {
@@ -656,15 +652,24 @@ int board_video_skip(void)
 	return 0;
 }
 
+#define RGB_BACKLIGHT_REVB 	IMX_GPIO_NR(4, 20)
+#define RGB_BACKLIGHT_GP 	IMX_GPIO_NR(2, 9)
+
 iomux_v3_cfg_t const backlight_pads[] = {
-	MX6_PAD_SD4_DAT1__GPIO_2_9 | MUX_PAD_CTRL(ENET_PAD_CTRL),
+	MX6_PAD_DI0_PIN4__GPIO_4_20 | MUX_PAD_CTRL(ENET_PAD_CTRL),
+	MX6_PAD_SD4_DAT1__GPIO_2_9  | MUX_PAD_CTRL(ENET_PAD_CTRL),
 };
 
+void board_set_backlight (int status)
+{
+	gpio_direction_output(RGB_BACKLIGHT_GP,   status);
+	gpio_direction_output(RGB_BACKLIGHT_REVB, status);
+}
+
 static void setup_iomux_backlight(void)
 {
 	imx_iomux_v3_setup_multiple_pads(backlight_pads,
 					 ARRAY_SIZE(backlight_pads));
-	gpio_direction_output(IMX_GPIO_NR(2, 9), 1);
 }
 
 static void setup_display(void)
@@ -674,12 +679,14 @@ static void setup_display(void)
 	struct iomuxc *iomux = (struct iomuxc *)IOMUXC_BASE_ADDR;
 	int reg;
 
+	setup_iomux_backlight();
+	board_set_backlight(1);
 	enable_ipu_clock();
 	imx_setup_hdmi();
+
 	/* Turn on LDB0,IPU,IPU DI0 clocks */
 	reg = __raw_readl(&mxc_ccm->CCGR3);
 	reg |=  (MXC_CCM_CCGR3_LDB_DI0_MASK | 0xffff); //TBD MM MP:
-	printf("CCGR3 = 0x%x\n", reg);
 	writel(reg, &mxc_ccm->CCGR3);
 
 	/* set PFD1_FRAC to 0x13 == 455 MHz (480*18)/0x13 */
diff --git a/boards.cfg b/boards.cfg
index bfcb35e..c852f4f 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -291,11 +291,13 @@ mx6slevkandroid          arm         armv7       mx6slevk            freescale
 mx6s_icore                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=256,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
 mx6s_icore_512MB              arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=512,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
 mx6dl_icore                   arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=512,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
+mx6dl1g_icore                   arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=1024,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
 mx6d_icore                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=512,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
 mx6q_icore                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=1024,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
 mx6s_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=256,SYS_NOSMP="nosmp"
 mx6s_icore_512MB_sd              arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
 mx6dl_icore_sd                   arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
+mx6dl1g_icore_sd                   arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=1024,SYS_NOSMP="nosmp"
 mx6d_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
 mx6q_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=1024,SYS_NOSMP="nosmp"
 mx6s_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore-rqs.dtb",DDR_MB=256,SYS_NOSMP="nosmp"
diff --git a/common/hush.c b/common/hush.c
index cc81c9c..61b3fe0 100644
--- a/common/hush.c
+++ b/common/hush.c
@@ -3222,6 +3222,16 @@ static int parse_stream_outer(struct in_str *inp, int flag)
 #ifndef __U_BOOT__
 	return 0;
 #else
+	#ifndef CONFIG_SYS_BOOT_NAND // Exluded first time if build for sd
+	static bool bFirstTime=true;
+	if(bFirstTime)
+		bFirstTime=false;
+	else
+		board_set_backlight(0);
+	#else
+	board_set_backlight(0);
+	#endif
+	
 	return (code != 0) ? 1 : 0;
 #endif /* __U_BOOT__ */
 }
diff --git a/common/main.c b/common/main.c
index 2e321ee..1507498 100644
--- a/common/main.c
+++ b/common/main.c
@@ -60,7 +60,7 @@
 DECLARE_GLOBAL_DATA_PTR;
 
 
-#define SHOW_ENGICAM_VERSION     "Version: Engicam Yocto U-Boot 1.00\n"
+#define SHOW_ENGICAM_VERSION     "Version: Engicam Yocto U-Boot 1.01\n"
 
 
 /*
@@ -400,7 +400,6 @@ static void process_fdt_options(const void *blob)
 }
 #endif /* CONFIG_OF_CONTROL */
 
-
 /****************************************************************************/
 
 void main_loop (void)
@@ -553,6 +552,7 @@ void main_loop (void)
 	debug ("### main_loop: bootcmd=\"%s\"\n", s ? s : "<UNDEFINED>");
 
 	if (bootdelay != -1 && s && !abortboot(bootdelay)) {
+	board_set_backlight(0);
 # ifdef CONFIG_AUTOBOOT_KEYED
 		int prev = disable_ctrlc(1);	/* disable Control C checking */
 # endif
@@ -563,7 +563,6 @@ void main_loop (void)
 		disable_ctrlc(prev);	/* restore Control C checking */
 # endif
 	}
-
 # ifdef CONFIG_MENUKEY
 	if (menukey == CONFIG_MENUKEY) {
 		s = getenv("menucmd");
diff --git a/include/common.h b/include/common.h
index 0cfa6a8..5a6329c 100644
--- a/include/common.h
+++ b/include/common.h
@@ -501,6 +501,7 @@ int board_late_init (void);
 int board_postclk_init (void); /* after clocks/timebase, before env/serial */
 int board_early_init_r (void);
 void board_poweroff (void);
+void board_set_backlight (int status);
 
 #if defined(CONFIG_SYS_DRAM_TEST)
 int testdram(void);
diff --git a/include/configs/mx6_icore_common.h b/include/configs/mx6_icore_common.h
index 1c2d6e2..b9b313b 100644
--- a/include/configs/mx6_icore_common.h
+++ b/include/configs/mx6_icore_common.h
@@ -152,7 +152,7 @@
 	 *     mtd3: 1M      (dtb)
 	 *     mtd4: left    (rootfs)
 	 */
-#ifdef CONFIG_MX6SOLO
+#if (defined(CONFIG_MX6SOLO) || defined(CONFIG_MX6DL))
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	EXTRA_ENV_SETTINGS_ICORE \
 	CONFIG_MFG_ENV_SETTINGS \
