--- a/include/configs/mx6_icore_common.h	2014-11-07 12:34:37.093275132 +0100
+++ b/include/configs/mx6_icore_common.h	2014-11-07 16:40:17.285453244 +0100
@@ -58,6 +58,18 @@
 
 #define CONFIG_MXC_UART
 
+/* UBI/UBI config options */
+#define CONFIG_CMD_FLASH
+#define CONFIG_MTD_DEVICE
+#define CONFIG_MTD_PARTITIONS
+#define CONFIG_CMD_MTDPARTS
+#define CONFIG_RBTREE
+#define CONFIG_CMD_UBI
+#define CONFIG_CMD_UBIFS
+#define CONFIG_LZO
+#define CONFIG_CMD_NAND
+#define CONFIG_CMD_NAND_TRIMFFS
+
 /* OCOTP Config */
 #define CONFIG_CMD_FUSE
 #ifdef CONFIG_CMD_FUSE

--- a/common/main.c	2014-11-10 10:28:22.000000000 +0100
+++ b/common/main.c	2014-11-10 16:14:37.152180852 +0100
@@ -60,7 +60,7 @@
 DECLARE_GLOBAL_DATA_PTR;
 
 
-#define SHOW_ENGICAM_VERSION     "Version: Engicam Yocto U-Boot 1.01\n"
+#define SHOW_ENGICAM_VERSION     "Version: Engicam Yocto U-Boot 1.02\n"
 
 
 /*
