diff --git a/board/engicam/mx6_icore/mx6_icore.c b/board/engicam/mx6_icore/mx6_icore.c
index 9ba6d04..6777dad 100644
--- a/board/engicam/mx6_icore/mx6_icore.c
+++ b/board/engicam/mx6_icore/mx6_icore.c
@@ -567,6 +567,26 @@ static struct display_info_t const displays[] = {{
 	.addr	= 0,
 	.pixfmt	= IPU_PIX_FMT_RGB666,
 	.detect	= NULL,
+	.enable	= enable_rgb,
+	.mode	= {
+		.name           = "LDB-WXGAI",
+		.refresh        = 60,
+		.xres           = 1280,
+		.yres           = 800,
+		.pixclock       = 14065,
+		.left_margin    = 40,
+		.right_margin   = 40,
+		.upper_margin   = 10,
+		.lower_margin   = 3,
+		.hsync_len      = 80,
+		.vsync_len      = 10,
+		.sync           = FB_SYNC_CLK_LAT_FALL,
+		.vmode          = FB_VMODE_NONINTERLACED
+} }, {
+	.bus	= -1,
+	.addr	= 0,
+	.pixfmt	= IPU_PIX_FMT_RGB666,
+	.detect	= NULL,
 	.enable	= NULL,
 	.mode	= {
 		.name           = "Hannstar-XGA",
@@ -608,7 +628,7 @@ int board_video_skip(void)
 {
 	int i;
 	int ret;
-	char const *panel = getenv("panel");
+	char const *panel = getenv("lcd_panel");
 
 	if (!panel) {
 		for (i = 0; i < ARRAY_SIZE(displays); i++) {
diff --git a/boards.cfg b/boards.cfg
index c852f4f..480d54c 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -300,10 +300,10 @@ mx6dl_icore_sd                   arm         armv7       mx6_icore       engicam
 mx6dl1g_icore_sd                   arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore.dtb",DDR_MB=1024,SYS_NOSMP="nosmp"
 mx6d_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
 mx6q_icore_sd                    arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore.dtb",DDR_MB=1024,SYS_NOSMP="nosmp"
-mx6s_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore-rqs.dtb",DDR_MB=256,SYS_NOSMP="nosmp"
-mx6dl_icore_rqs               arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore-rqs.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
-mx6d_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6q.cfg,MX6D,DEFAULT_FDT_FILE="imx6q-icore-rqs.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
-mx6q_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore-rqs.dtb",DDR_MB=1024,SYS_NOSMP="nosmp"
+mx6s_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-icore-rqs.dtb",DDR_MB=256,SYS_BOOT_EMMC,SYS_NOSMP="nosmp"
+mx6dl_icore_rqs               arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore-rqs.dtb",DDR_MB=512,SYS_BOOT_EMMC,SYS_NOSMP="nosmp"
+mx6d_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6q.cfg,MX6D,DEFAULT_FDT_FILE="imx6q-icore-rqs.dtb",DDR_MB=512,SYS_BOOT_EMMC,SYS_NOSMP="nosmp"
+mx6q_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore-rqs.dtb",DDR_MB=1024,SYS_BOOT_EMMC,SYS_NOSMP="nosmp"
 eco5pk                       arm         armv7       eco5pk              8dtech         omap3
 nitrogen6dl                  arm         armv7       nitrogen6x          boundary       mx6		nitrogen6x:IMX_CONFIG=board/boundary/nitrogen6x/nitrogen6dl.cfg,MX6DL,DDR_MB=1024
 nitrogen6dl2g                arm         armv7       nitrogen6x          boundary       mx6		nitrogen6x:IMX_CONFIG=board/boundary/nitrogen6x/nitrogen6dl2g.cfg,MX6DL,DDR_MB=2048
diff --git a/include/configs/common_parameter.h b/include/configs/common_parameter.h
index 6bb1505..2b2e778 100644
--- a/include/configs/common_parameter.h
+++ b/include/configs/common_parameter.h
@@ -17,11 +17,13 @@
 	#define CONFIG_ETHADDR		60:aa:60:aa:60:aa
 
 	#if defined(CONFIG_SYS_BOOT_NAND)
-		#define CONFIG_BOOTCMD	"bootcmd=run bootcmd_ubi\0"
+		#define CONFIG_BOOTCMD		"bootcmd=run bootcmd_ubi\0"
 	#elif defined(CONFIG_SYS_BOOT_SATA)
-		#define CONFIG_BOOTCMD	"bootcmd=run bootcmd_ubi\0"
+		#define CONFIG_BOOTCMD		"bootcmd=run bootcmd_sata\0"
+	#elif defined(CONFIG_SYS_BOOT_EMMC)
+		#define CONFIG_BOOTCMD		"bootcmd=run bootcmd_emmc\0"
 	#else
-		#define CONFIG_BOOTCMD	"bootcmd=run bootcmd_mmc\0"
+		#define CONFIG_BOOTCMD		"bootcmd=run bootcmd_mmc\0"
 	#endif
 
 	#ifdef CONFIG_MX6SOLO
@@ -37,13 +39,15 @@
 		"netdev=eth0\0" 			\
 		"ethprime=FEC0\0" 			\
 		"lcd_panel=Amp-WD\0" 			\
-		"nfsroot=nfs_icore\0"			\
+		"nfsroot=/nfs_icore\0"			\
+		"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0"\
+		CONFIG_BOOTCMD				\
 		"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0"											\
 		"bootargs=set bootargs console=" CONFIG_CONSOLE_DEV ",115200" EXTRA_OPTION_SOLO "engi_board=${board} video=${video_type},${lcd_panel} mtdparts=gpmi-nand:4m(boot),8m(kernel),1m(dtb),-(rootfs)\0"		\
 		"bootargs_net=setenv bootargs ${bootargs} root=/dev/nfs ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" 						\
 		"bootargs_mmc=setenv bootargs ${bootargs} root=/dev/mmcblk0p2 rootwait rw\0" 										\
 		"bootcmd_mmc=run bootargs;  run bootargs_mmc; run loadfdt; fatload mmc 0:1 ${loadaddr} uImage; bootm ${loadaddr} - ${fdt_addr};\0"			\
-		"bootcmd_net=run bootargs; run bootargs_net; tftp uImage; bootm\0" \
+		"bootcmd_net=run bootargs; run bootargs_net; tftp uImage; tftp ${fdt_addr} uImage.dtb; bootm ${loadaddr} - ${fdt_addr}\0"				\
 
 	/* Customized parameter
 	 * Customized parameter for SODIMM iCore modules
@@ -53,7 +57,6 @@
 		"bootcmd_ubi=run bootargs; run bootargs_ubi; nand read ${loadaddr} 0x400000 0x700000;nand read ${fdt_addr} 0xc00000 0x100000;bootm ${loadaddr} - ${fdt_addr}\0" 		\
 		"bootargs_ubi=setenv bootargs ${bootargs} ubi.mtd=3 root=ubi0:rootfs rootfstype=ubifs\0"				\
 		"video_type=mxcfb0:dev=lcd\0"		\
-		CONFIG_BOOTCMD 				\
 		"erase=nand erase 1c0000 4000\0"	\
 		"board=SK.RES\0"			\
 
@@ -65,7 +68,6 @@
 		"bootcmd_emmc=run bootargs; run bootargs_emmc; mmc dev 1; mmc read ${loadaddr} 0x800 0x2000;bootm\0"					\
 		"bootargs_emmc=setenv bootargs ${bootargs} root=/dev/mmcblk0p1 rootwait rw\0" 								\
 		"video_type=mxcfb0:dev=ldb,LDB-XGA,if=RGB666 video=mxcfb0:dev=hdmi,1920x1080M@60,if=RGB24\0"						\
-		"bootcmd=run bootcmd_emmc\0" 		\
 		"erase=mmc dev 1; mmc erase 400 1\0"	\
 		"board=SK.Q7\0"				\
 
diff --git a/include/configs/mx6_icore_common.h b/include/configs/mx6_icore_common.h
index b786594..8749dda 100644
--- a/include/configs/mx6_icore_common.h
+++ b/include/configs/mx6_icore_common.h
@@ -167,15 +167,11 @@
 #if (defined(CONFIG_MX6SOLO) || defined(CONFIG_MX6DL))
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	EXTRA_ENV_SETTINGS_ICORE \
-	CONFIG_MFG_ENV_SETTINGS \
-	"fdt_file=imx6dl-icore.dtb\0"		\
 	"fdt_addr=0x18000000\0" \
 	"fdt_high=0xffffffff\0"
 #else
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	EXTRA_ENV_SETTINGS_ICORE \
-	CONFIG_MFG_ENV_SETTINGS \
-	"fdt_file=imx6q-icore.dtb\0"		\
 	"fdt_addr=0x18000000\0" \
 	"fdt_high=0xffffffff\0"
 #endif
diff --git a/include/configs/mx6_icore_rqs_common.h b/include/configs/mx6_icore_rqs_common.h
index ca9f899..1840893 100644
--- a/include/configs/mx6_icore_rqs_common.h
+++ b/include/configs/mx6_icore_rqs_common.h
@@ -46,6 +46,7 @@
 #define CONFIG_DISPLAY_BOARDINFO
 
 #include <asm/arch/imx-regs.h>
+#include "common_parameter.h"
 
 #define CONFIG_CMDLINE_TAG
 #define CONFIG_SETUP_MEMORY_TAGS
@@ -152,7 +153,7 @@
 		"bootcmd_mfg=run mfgtool_args;bootm ${loadaddr} ${initrd_addr} ${fdt_addr};\0" \
 
 
-#if defined(CONFIG_SYS_BOOT_NAND)
+#if defined(CONFIG_SYS_BOOT_EMMC)
 	/*
 	 * The dts also enables the WEIN NOR which is mtd0.
 	 * So the partions' layout for NAND is:
@@ -161,16 +162,17 @@
 	 *     mtd3: 1M      (dtb)
 	 *     mtd4: left    (rootfs)
 	 */
+#if (defined(CONFIG_MX6SOLO) || defined(CONFIG_MX6DL))
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	CONFIG_MFG_ENV_SETTINGS \
+	EXTRA_ENV_SETTINGS_RQS \
 	"fdt_addr=0x18000000\0" \
-	"fdt_high=0xffffffff\0"	  \
-	"bootargs=console=" CONFIG_CONSOLE_DEV ",115200 ubi.mtd=4 "  \
-		"root=ubi0:rootfs rootfstype=ubifs "		     \
-		"mtdparts=gpmi-nand:4m(boot),8m(kernel),1m(dtb),-(rootfs)\0"\
-	"bootcmd=nand read ${loadaddr} 0x400000 0x700000;"\
-		"nand read ${fdt_addr} 0xc00000 0x100000;"\
-		"bootm ${loadaddr} - ${fdt_addr}\0"
+	"fdt_high=0xffffffff\0"
+#else
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	EXTRA_ENV_SETTINGS_RQS \
+	"fdt_addr=0x18000000\0" \
+	"fdt_high=0xffffffff\0"
+#endif
 
 #elif defined(CONFIG_SYS_BOOT_SATA)
 
