From a6b589ab6a8511302eadc4523c27ce342f73b271 Mon Sep 17 00:00:00 2001
From: Your Name <Your Email>
Date: Fri, 11 Sep 2015 12:36:24 +0200
Subject: [PATCH] fix icorem6 fdt varable

---
 include/configs/icorem6.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/icorem6.h b/include/configs/icorem6.h
index b3f9c6f..888d7e9 100644
--- a/include/configs/icorem6.h
+++ b/include/configs/icorem6.h
@@ -40,9 +40,9 @@
 #define CONFIG_CMDLINE_EDITING
 
 #if defined(CONFIG_MX6Q)
-#define CONFIG_DEFAULT_FDT_FILE	"icorem6q.dtb"
+#define CONFIG_DEFAULT_FDT_FILE	"imx6q-icore.dtb"
 #elif defined(CONFIG_MX6DL)
-#define CONFIG_DEFAULT_FDT_FILE	"icorem6dl.dtb"
+#define CONFIG_DEFAULT_FDT_FILE	"imx6dl-icore.dtb"
 #endif
 #define PHYS_SDRAM_SIZE		(CONFIG_DDR_MB * 1024 * 1024)
 
-- 
1.7.9.5

