From 7770a0fcd3ff8e137a6ebcac45b988c254fb9185 Mon Sep 17 00:00:00 2001
From: "matteo.lisi" <matteo.lisi@engicam.com>
Date: Mon, 26 Oct 2015 17:25:02 +0100
Subject: [PATCH] fix fdt_address for icorem6sx

---
 include/configs/icore_mx6sx.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/icore_mx6sx.h b/include/configs/icore_mx6sx.h
index dd91111..85f6907 100644
--- a/include/configs/icore_mx6sx.h
+++ b/include/configs/icore_mx6sx.h
@@ -18,7 +18,7 @@
 #include <linux/sizes.h>
 #include "mx6_common.h"
 #include <asm/imx-common/gpio.h>
-#define CONFIG_DEFAULT_FDT_FILE "imx6sx-smarcore.dtb"
+#define CONFIG_DEFAULT_FDT_FILE "imx6sx-icore.dtb"
 #include "common_parameter.h"
 
 #define CONFIG_MX6
@@ -175,6 +175,7 @@
 	
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	EXTRA_ENV_SETTINGS_ICORE \
+       "fdt_addr=0x83000000 \0" \
 	"fdt_high=0xffffffff\0"
 
 #define CONFIG_BOOTCOMMAND \
-- 
1.9.1

