diff -uNr wpa_supplicant-0.7.3/src/common/version.h wpa_supplicant-0.7.3_mod/src/common/version.h
--- wpa_supplicant-0.7.3/src/common/version.h	2010-09-07 16:43:39.000000000 +0100
+++ wpa_supplicant-0.7.3_mod/src/common/version.h	2011-12-22 12:35:01.214036820 +0000
@@ -1,6 +1,6 @@
 #ifndef VERSION_H
 #define VERSION_H
 
-#define VERSION_STR "0.7.3"
+#define VERSION_STR "0.7.3-csr"
 
 #endif /* VERSION_H */
diff -uNr wpa_supplicant-0.7.3/src/drivers/driver_wext.c wpa_supplicant-0.7.3_mod/src/drivers/driver_wext.c
--- wpa_supplicant-0.7.3/src/drivers/driver_wext.c	2010-09-07 16:43:39.000000000 +0100
+++ wpa_supplicant-0.7.3_mod/src/drivers/driver_wext.c	2011-12-22 12:37:00.376098412 +0000
@@ -399,7 +399,7 @@
 {
 	union wpa_event_data data;
 
-	if (drv->assoc_req_ies == NULL && drv->assoc_resp_ies == NULL)
+	if (drv->assoc_req_ies == NULL && drv->assoc_resp_ies == NULL && drv->scan_gen_ies == NULL)
 		return;
 
 	os_memset(&data, 0, sizeof(data));
@@ -411,13 +411,54 @@
 		data.assoc_info.resp_ies = drv->assoc_resp_ies;
 		data.assoc_info.resp_ies_len = drv->assoc_resp_ies_len;
 	}
-
+	if (drv->scan_gen_ies) {
+		data.assoc_info.beacon_ies = drv->scan_gen_ies;
+		data.assoc_info.beacon_ies_len = drv->scan_gen_ies_len;
+	}
+    
 	wpa_supplicant_event(drv->ctx, EVENT_ASSOCINFO, &data);
 
 	os_free(drv->assoc_req_ies);
 	drv->assoc_req_ies = NULL;
 	os_free(drv->assoc_resp_ies);
 	drv->assoc_resp_ies = NULL;
+	os_free(drv->scan_gen_ies);
+	drv->scan_gen_ies = NULL;
+}
+
+static void wpa_driver_wext_update_beacon_ies(struct wpa_driver_wext_data *drv,
+                                              const u8 *bssid)
+{
+    struct wpa_scan_results *scan_res;
+    struct wpa_scan_res *r;
+    int i;
+    const u8 *ie_pos;
+
+    scan_res = wpa_driver_wext_get_scan_results(drv);
+    if (scan_res == NULL) {
+        wpa_printf(MSG_DEBUG, "wpa_driver_wext_update_beacon_ies: failed to get scan results");
+        return;
+    }
+
+    /* Search for the current BSSID */
+    for (i = 0; i < scan_res->num; i++) {
+        r = scan_res->res[i];
+        if (os_memcmp(r->bssid, bssid, ETH_ALEN) != 0)
+            continue;
+
+        ie_pos = (const u8 *) (r + 1);
+
+        wpa_hexdump(MSG_DEBUG, "wpa_driver_wext_update_beacon_ies:", ie_pos,
+                    r->ie_len);
+        os_free(drv->scan_gen_ies);
+        drv->scan_gen_ies = os_malloc(r->ie_len);
+        if (drv->scan_gen_ies == NULL) {
+            drv->scan_gen_ies_len = 0;
+            return;
+        }
+        os_memcpy(drv->scan_gen_ies, ie_pos, r->ie_len);
+        drv->scan_gen_ies_len = r->ie_len;
+    }
 }
 
 
@@ -474,6 +515,7 @@
 						     NULL);
 			
 			} else {
+                wpa_driver_wext_update_beacon_ies(drv, (const u8*)iwe->u.ap_addr.sa_data);
 				wpa_driver_wext_event_assoc_ies(drv);
 				wpa_supplicant_event(drv->ctx, EVENT_ASSOC,
 						     NULL);
@@ -1044,8 +1086,8 @@
 			       struct wext_scan_data *res)
 {
 	res->res.qual = iwe->u.qual.qual;
-	res->res.noise = iwe->u.qual.noise;
-	res->res.level = iwe->u.qual.level;
+	res->res.noise = ( iwe->u.qual.noise > 63 ) ? iwe->u.qual.noise - 0x100 : iwe->u.qual.noise;
+	res->res.level = ( iwe->u.qual.level > 63 ) ? iwe->u.qual.level - 0x100 : iwe->u.qual.level;
 	if (iwe->u.qual.updated & IW_QUAL_QUAL_INVALID)
 		res->res.flags |= WPA_SCAN_QUAL_INVALID;
 	if (iwe->u.qual.updated & IW_QUAL_LEVEL_INVALID)
diff -uNr wpa_supplicant-0.7.3/src/drivers/driver_wext.h wpa_supplicant-0.7.3_mod/src/drivers/driver_wext.h
--- wpa_supplicant-0.7.3/src/drivers/driver_wext.h	2010-09-07 16:43:39.000000000 +0100
+++ wpa_supplicant-0.7.3_mod/src/drivers/driver_wext.h	2011-12-22 12:24:04.669017306 +0000
@@ -30,6 +30,8 @@
 	size_t assoc_req_ies_len;
 	u8 *assoc_resp_ies;
 	size_t assoc_resp_ies_len;
+    u8 *scan_gen_ies;
+    size_t scan_gen_ies_len;
 	struct wpa_driver_capa capa;
 	int has_capability;
 	int we_version_compiled;
diff -uNr wpa_supplicant-0.7.3/src/rsn_supp/pmksa_cache.c wpa_supplicant-0.7.3_mod/src/rsn_supp/pmksa_cache.c
--- wpa_supplicant-0.7.3/src/rsn_supp/pmksa_cache.c	2010-09-07 16:43:39.000000000 +0100
+++ wpa_supplicant-0.7.3_mod/src/rsn_supp/pmksa_cache.c	2011-12-22 12:24:04.679908942 +0000
@@ -63,6 +63,7 @@
 	os_get_time(&now);
 	while (pmksa->pmksa && pmksa->pmksa->expiration <= now.sec) {
 		struct rsn_pmksa_cache_entry *entry = pmksa->pmksa;
+		wpa_sm_remove_pmkid(pmksa->sm, entry->aa, entry->pmkid);
 		pmksa->pmksa = entry->next;
 		wpa_printf(MSG_DEBUG, "RSN: expired PMKSA cache entry for "
 			   MACSTR, MAC2STR(entry->aa));
