From 26457384ae1c980766b250024b8be018094e6fb1 Mon Sep 17 00:00:00 2001
From: "matteo.lisi" <matteo.lisi@engicam.com>
Date: Fri, 16 Oct 2015 16:41:22 +0200
Subject: [PATCH] ini

---
 tools/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/Makefile b/tools/Makefile
index 4bbb153..f2ffd28 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -186,7 +186,7 @@ LOGO-$(CONFIG_VIDEO_LOGO) += $(LOGO_DATA_H)
 
 # Generic logo
 ifeq ($(LOGO_BMP),)
-LOGO_BMP= $(srctree)/$(src)/logos/denx.bmp
+LOGO_BMP= $(srctree)/$(src)/logos/Engicam.bmp
 
 # Use board logo and fallback to vendor
 ifneq ($(wildcard $(srctree)/$(src)/logos/$(BOARD).bmp),)
-- 
1.9.1

