From be4cf026fe411fee52bf77ecde4225e5a08a06f1 Mon Sep 17 00:00:00 2001
From: Engicam <support@engicam.com>
Date: Tue, 17 Jun 2014 15:35:37 +0200
Subject: [PATCH 8/9] ADV7180 video in on i.Core (part 2)

---
 drivers/media/platform/mxc/capture/adv7180.c |   12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/adv7180.c b/drivers/media/platform/mxc/capture/adv7180.c
index 8e7f3e1..0257e2f 100644
--- a/drivers/media/platform/mxc/capture/adv7180.c
+++ b/drivers/media/platform/mxc/capture/adv7180.c
@@ -33,18 +33,6 @@
 #include <media/v4l2-int-device.h>
 #include "mxc_v4l2_capture.h"
 
-#undef dev_dbg
-#define dev_dbg(dev, format, arg...)		\
-	printk(format, ##arg)
-
-#undef dev_warn
-#define dev_warn(dev, format, arg...)		\
-	printk(format, ##arg)
-
-#undef dev_err
-#define dev_err(dev, format, arg...)		\
-	printk(format, ##arg)
-
 #define ADV7180_VOLTAGE_ANALOG               1800000
 #define ADV7180_VOLTAGE_DIGITAL_CORE         1800000
 #define ADV7180_VOLTAGE_DIGITAL_IO           3300000
-- 
1.7.9.5

