diff --git a/drivers/video/mxc/mxc_lcdif.c b/drivers/video/mxc/mxc_lcdif.c
index fff54ba..b8262a1 100644
--- a/drivers/video/mxc/mxc_lcdif.c
+++ b/drivers/video/mxc/mxc_lcdif.c
@@ -111,7 +111,12 @@ static struct fb_videomode lcdif_modedb[] = {
 	FB_SYNC_CLK_LAT_FALL | FB_SYNC_COMP_HIGH_ACT | FB_SYNC_VERT_HIGH_ACT,
 	FB_VMODE_NONINTERLACED
          },
-
+	 {
+     	/* GO70Y2-L01 800x480 @ 60 Hz 7'' */
+     	"GO70Y2-L01" , 60, 800, 480, 30000, 30, 30, 5, 5, 64, 20,
+     	FB_SYNC_CLK_LAT_FALL,
+     	FB_VMODE_NONINTERLACED,
+     	0,},
 };
 static int lcdif_modedb_sz = ARRAY_SIZE(lcdif_modedb);
 
