diff --git a/arch/arm/boot/dts/imx6dl-icore.dts b/arch/arm/boot/dts/imx6dl-icore.dts
index 5e75896..2a9b36e 100644
--- a/arch/arm/boot/dts/imx6dl-icore.dts
+++ b/arch/arm/boot/dts/imx6dl-icore.dts
@@ -97,6 +97,11 @@
 		mclk_source = <0>;
 		cvbs = <1>;
 	};
+	
+	pcf8563: rtc@51 {
+		compatible = "pcf8563";
+		reg = <0x51>;
+	};
 };
 
 &iomuxc {
diff --git a/arch/arm/boot/dts/imx6q-icore.dts b/arch/arm/boot/dts/imx6q-icore.dts
index 2f7ce37..1560af0 100644
--- a/arch/arm/boot/dts/imx6q-icore.dts
+++ b/arch/arm/boot/dts/imx6q-icore.dts
@@ -59,7 +59,6 @@
 		compatible = "fsl,imx6-hdmi-i2c";
 		reg = <0x50>;
 	};
-
 };
 
  &i2c3 {
@@ -100,6 +99,12 @@
 		mclk_source = <0>;
 		cvbs = <1>;
 	};
+	
+	pcf8563: rtc@51 {
+		compatible = "pcf8563";
+		reg = <0x51>;
+	};
+
 };
 
 &iomuxc {
diff --git a/drivers/video/mxc/ldb.c b/drivers/video/mxc/ldb.c
index 6e7c685..004ee51 100644
--- a/drivers/video/mxc/ldb.c
+++ b/drivers/video/mxc/ldb.c
@@ -195,6 +195,16 @@ static struct fb_videomode ldb_modedb[] = {
 	FB_SYNC_CLK_LAT_FALL | FB_SYNC_COMP_HIGH_ACT | FB_SYNC_VERT_HIGH_ACT,
 	FB_VMODE_NONINTERLACED
          },
+	{
+	/* 1024x768  */
+        "LVDS-G150" , 65, 1024, 768, 15384,
+	40, 40,
+	5, 5,
+	136, 6,
+	0,
+        FB_VMODE_NONINTERLACED,
+	FB_MODE_IS_DETAILED,
+        },
 
 };
 static int ldb_modedb_sz = ARRAY_SIZE(ldb_modedb);
diff --git a/drivers/video/mxc/mxc_lcdif.c b/drivers/video/mxc/mxc_lcdif.c
index 81463f9..f2cbdfa 100644
--- a/drivers/video/mxc/mxc_lcdif.c
+++ b/drivers/video/mxc/mxc_lcdif.c
@@ -117,6 +117,16 @@ static struct fb_videomode lcdif_modedb[] = {
      	FB_SYNC_CLK_LAT_FALL,
      	FB_VMODE_NONINTERLACED,
      	0,},
+	{
+	/* 1024x768  */
+        "LVDS-G150" , 65, 1024, 768, 15384,
+	40, 40,
+	5, 5,
+	136, 6,
+	0,
+        FB_VMODE_NONINTERLACED,
+	FB_MODE_IS_DETAILED,
+        },
 };
 static int lcdif_modedb_sz = ARRAY_SIZE(lcdif_modedb);
 
