diff --git a/arch/arm/boot/dts/imx6dl.dtsi b/arch/arm/boot/dts/imx6dl.dtsi
index 3d19bfa..df0c9b5 100644
--- a/arch/arm/boot/dts/imx6dl.dtsi
+++ b/arch/arm/boot/dts/imx6dl.dtsi
@@ -23,13 +23,11 @@
 			next-level-cache = <&L2>;
 			operating-points = <
 				/* kHz    uV */
-				996000  1275000
 				792000  1175000
 				396000  1075000
 			>;
 			fsl,soc-operating-points = <
 				/* ARM kHz      SOC-PU uV */
-				996000            1175000
 				792000            1175000
 				396000            1175000
 			>;
diff --git a/arch/arm/boot/dts/imx6q.dtsi b/arch/arm/boot/dts/imx6q.dtsi
index 5806312..ea459a7 100644
--- a/arch/arm/boot/dts/imx6q.dtsi
+++ b/arch/arm/boot/dts/imx6q.dtsi
@@ -27,17 +27,11 @@
 			next-level-cache = <&L2>;
 			operating-points = <
 				/* kHz    uV */
-				1200000 1275000
-				996000  1250000
-				852000  1250000
 				792000  1150000
 				396000  975000
 			>;
 			fsl,soc-operating-points = <
 				/* ARM kHz  SOC-PU uV */
-				1200000       1275000
-				996000        1250000
-				852000        1250000
 				792000        1175000
 				396000        1175000
 			>;

