From 40fdb3bbb3436ccfdc4065b2b6ce76557d498449 Mon Sep 17 00:00:00 2001
From: Your Name <Your Email>
Date: Tue, 6 Oct 2015 12:38:07 +0200
Subject: [PATCH 21/27] thermale-temperature-change

---
 drivers/thermal/imx_thermal.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/thermal/imx_thermal.c b/drivers/thermal/imx_thermal.c
index 446f1cf..ba7936d 100644
--- a/drivers/thermal/imx_thermal.c
+++ b/drivers/thermal/imx_thermal.c
@@ -58,7 +58,7 @@
  * It defines the temperature in millicelsius for passive trip point
  * that will trigger cooling action when crossed.
  */
-#define IMX_TEMP_PASSIVE		85000
+#define IMX_TEMP_PASSIVE		100000
 #define IMX_TEMP_PASSIVE_COOL_DELTA     10000
 
 #define IMX_POLLING_DELAY		2000 /* millisecond */
-- 
1.7.9.5

