From a03bfc9330948acc2f23c06f910f1cb7005b4582 Mon Sep 17 00:00:00 2001
From: Your Name <Your Email>
Date: Tue, 6 Oct 2015 13:43:04 +0200
Subject: [PATCH 26/27] ofloco-bugfix-touchscreen

---
 drivers/input/touchscreen/stmpe-ts.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/stmpe-ts.c b/drivers/input/touchscreen/stmpe-ts.c
index 03c8ec9..666e62e 100644
--- a/drivers/input/touchscreen/stmpe-ts.c
+++ b/drivers/input/touchscreen/stmpe-ts.c
@@ -234,7 +234,7 @@ static irqreturn_t stmpe_ts_handler(int irq, void *data)
 
 	
 	input_report_abs(ts->idev, ABS_X, x);
-	input_report_abs(ts->idev, ABS_Y, y);
+	input_report_abs(ts->idev, ABS_Y, 4096 - y);
 	input_report_abs(ts->idev, ABS_PRESSURE, z);
 	input_report_key(ts->idev, BTN_TOUCH, 1);
 	input_sync(ts->idev);
-- 
1.7.9.5

