diff --git a/arch/arm/boot/dts/imx6ul-gea.dts b/arch/arm/boot/dts/imx6ul-gea.dts
index df1e486..965452b 100644
--- a/arch/arm/boot/dts/imx6ul-gea.dts
+++ b/arch/arm/boot/dts/imx6ul-gea.dts
@@ -407,6 +407,7 @@
 				MX6UL_PAD_ENET2_TX_EN__ENET2_TX_EN	0x1b0b0
 				MX6UL_PAD_ENET2_TX_DATA0__ENET2_TDATA00	0x1b0b0
 				MX6UL_PAD_ENET2_TX_DATA1__ENET2_TDATA01	0x1b0b0
+				MX6UL_PAD_GPIO1_IO05__ENET2_REF_CLK2	0x4001b031
 			>;
 		};
 
