diff --git a/boards.cfg b/boards.cfg
index 480d54c..29637f6 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -304,6 +304,10 @@ mx6s_icore_rqs                arm         armv7       mx6_icore_rqs       engica
 mx6dl_icore_rqs               arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-icore-rqs.dtb",DDR_MB=512,SYS_BOOT_EMMC,SYS_NOSMP="nosmp"
 mx6d_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6q.cfg,MX6D,DEFAULT_FDT_FILE="imx6q-icore-rqs.dtb",DDR_MB=512,SYS_BOOT_EMMC,SYS_NOSMP="nosmp"
 mx6q_icore_rqs                arm         armv7       mx6_icore_rqs       engicam      mx6	mx6_icore_rqs:IMX_CONFIG=board/engicam/mx6_icore_rqs/mx6q.cfg,MX6Q,DEFAULT_FDT_FILE="imx6q-icore-rqs.dtb",DDR_MB=1024,SYS_BOOT_EMMC,SYS_NOSMP="nosmp"
+mx6s_example                  arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-example.dtb",DDR_MB=256,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
+mx6s_example_sd               arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6solo.cfg,MX6SOLO,DEFAULT_FDT_FILE="imx6dl-example.dtb",DDR_MB=256,SYS_NOSMP="nosmp"
+mx6dl_example                 arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-example.dtb",DDR_MB=512,SYS_BOOT_NAND,SYS_NOSMP="nosmp"
+mx6dl_example_sd              arm         armv7       mx6_icore       engicam      mx6		mx6_icore:IMX_CONFIG=board/engicam/mx6_icore/mx6dl.cfg,MX6DL,DEFAULT_FDT_FILE="imx6dl-example.dtb",DDR_MB=512,SYS_NOSMP="nosmp"
 eco5pk                       arm         armv7       eco5pk              8dtech         omap3
 nitrogen6dl                  arm         armv7       nitrogen6x          boundary       mx6		nitrogen6x:IMX_CONFIG=board/boundary/nitrogen6x/nitrogen6dl.cfg,MX6DL,DDR_MB=1024
 nitrogen6dl2g                arm         armv7       nitrogen6x          boundary       mx6		nitrogen6x:IMX_CONFIG=board/boundary/nitrogen6x/nitrogen6dl2g.cfg,MX6DL,DDR_MB=2048
