From becf17ec8b97937dd2323e7953830c3a37ea5c32 Mon Sep 17 00:00:00 2001
From: Mirko Ardinghi <mirko.ardinghi@engicam.com>
Date: Wed, 18 Sep 2019 16:33:32 +0200
Subject: [PATCH] fix

---
 arch/arm64/kernel/process.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/kernel/process.c b/arch/arm64/kernel/process.c
index 9e773732520c..ee612e57acba 100644
--- a/arch/arm64/kernel/process.c
+++ b/arch/arm64/kernel/process.c
@@ -157,6 +157,9 @@ void machine_restart(char *cmd)
 	if (efi_enabled(EFI_RUNTIME_SERVICES))
 		efi_reboot(reboot_mode, NULL);
 
+  //[ENG] force using do_kernel_restart function
+  arm_pm_restart=0; 
+  
 	/* Now call the architecture specific reboot code. */
 	if (arm_pm_restart)
 		arm_pm_restart(reboot_mode, cmd);
-- 
2.20.1

